.github/workflows/migration-tests.yml

```diff
@@ -0,0 +1,111 @@
+name: 'Migration Tests'
+
+on:
+  workflow_dispatch: {} # workflow can be run manually
+  workflow_call: # workflow runs after code-health
+    inputs:
+      parent-event-name:
+        required: true
+        type: string
+  pull_request: # you can run a specic job in your PR using GitHub labels
+    types: [ labeled ]
+
+jobs:  
+  change-detection:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+      repository-projects: read
+    outputs:
+      project: ${{ steps.filter.outputs.project }}
+      config: ${{ steps.filter.outputs.config }}
+    steps:
+    - uses: actions/checkout@v4
+      if: ${{ github.event_name == 'push' ||  github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' || inputs.parent-event-name == 'release' }}
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        filters: |
+          project:
+            - 'mongodbatlas/fw_data_source_mongodbatlas_project_ip_access_list*.go'
+            - 'mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list*.go'
+          config:
+            - 'mongodbatlas/fw_data_source_mongodbatlas_alert_configuration*.go'
+            - 'mongodbatlas/fw_data_source_mongodbatlas_database_user*.go'
+            - 'mongodbatlas/fw_data_source_mongodbatlas_atlas_user*.go'
+            - 'mongodbatlas/fw_resource_mongodbatlas_alert_configuration*.go'
+            - 'mongodbatlas/fw_resource_mongodbatlas_database_user*.go'   
+
+  project: 
+    needs: [ change-detection ]
+    if: ${{ needs.change-detection.outputs.project == 'true' || github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' || github.event.label.name == 'run-testacc-migration' || github.event.label.name == 'run-testacc-project-migration' || inputs.parent-event-name == 'release' }}
```

Comment:
i know we have this in other gh actions but I wonder if we could make this easier or have some functions or something

Reply:
perhaps we could start fixing this file and then a follow-up to change the others?

Reply:
@lantoli I couldn't find a way to make this reusable or use a function, is there any example or reference you could share?

Reply:
I'm not sure about the best way, but for instance what about creating a variable in the workflow and use the variable? something like:

on: ...

env:
   hasToTrigger: ${{ needs.change-detection.outputs.config == 'true' || github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' || github.event.label.name == 'run-testacc-migration' || github.event.label.name == 'run-testacc-config-migration' || inputs.parent-event-name == 'release' }}

job...

and use:

if: ${{ env.hasToTrigger }}

Reply:
or use it for the common part and then:
if: ${{ env.hasToTrigger }} || specific conditions here...

Reply:
The thing is we need to pass arguments here, because github.event.label.name will be different, so we might be able to do this partially but not for entire condition

Reply:
since this doesn't hold a static value, rather holds a condition, i don't believe this is possible to do via a variable as well

Reply:
Addressed, see comment https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1565#discussion_r1372020123